zenodo/base/views.py

```diff
@@ -55,6 +55,7 @@ def register_menu_items():
 
     item = current_menu.submenu('main.browse')
     item.register('', _('Browse'), order=3)
+    item._active_when = lambda: request.endpoint.startswith("search.collection")
```

Comment:
You should be able to provide in  ``item.register(active_when=...)`` (see https://github.com/inveniosoftware/flask-menu/blob/master/flask_menu/__init__.py#L75)